org-timeline.el

```diff
@@ -147,25 +188,55 @@ Return new copy of STRING."
+                          (string-to-number (format-time-string "%M"))))
+         (current-offset (/ (- current-time start-offset) 10))
+         (slotline (org-timeline--add-elapsed-face
+                    "|     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |"
+                    current-offset))
+         (hourline (org-timeline--add-elapsed-face
+                    (concat "|"
+                            (mapconcat (lambda (x) (format "%02d:00" (mod x 24)))
+                                       (number-sequence org-timeline-start-hour (+ org-timeline-start-hour 23))
+                                       "|")
+                            "|")
+                    current-offset))
+         (timeline (concat hourline "\n" slotline))
+         (tasks (org-timeline--list-tasks)))
     (cl-labels ((get-start-pos (current-line beg) (+ 1 (* current-line (1+ (length slotline))) (/ (- beg start-offset) 10)))
                 (get-end-pos (current-line end) (+ 1 (* current-line (1+ (length slotline))) (/ (- end start-offset) 10))))
       (let ((current-line 1))
         (with-temp-buffer
           (insert timeline)
           (-each tasks
-            (-lambda ((beg end face))
+            (-lambda ((beg end face txt line))
               (while (get-text-property (get-start-pos current-line beg) 'org-timeline-occupied)
                 (cl-incf current-line)
                 (when (> (get-start-pos current-line beg) (point-max))
                   (save-excursion
                     (goto-char (point-max))
                     (insert "\n" slotline))))
               (let ((start-pos (get-start-pos current-line beg))
-                    (end-pos (get-end-pos current-line end)))
-                (put-text-property start-pos end-pos 'font-lock-face face)
-                (put-text-property start-pos end-pos 'org-timeline-occupied t))
+                    (end-pos (get-end-pos current-line end))
+                    (move-to-task-map (make-sparse-keymap))
+                    (props (list 'font-lock-face face
+                                 'org-timeline-occupied t
+                                 'mouse-face 'highlight
+                                 'txt txt
+                                 'help-echo (lambda (w obj pos)
+                                              (org-timeline--hover-info w txt)
+                                              txt) ;; the lambda will be called on block hover
+                                 'org-timeline-task-line line
+                                  'cursor-sensor-functions '(org-timeline--display-info))))
+                (define-key move-to-task-map [mouse-1] 'org-timeline--move-to-task)
+                (put-text-property start-pos end-pos 'keymap move-to-task-map)
+                (add-text-properties start-pos end-pos props))
```

Comment:
The keymap can be added to the `props` list, this will let us remove one function call.